javav2/example_code/iotsitewise/src/main/java/com/example/iotsitewise/scenario/CloudFormationHelper.java

```diff
@@ -0,0 +1,134 @@
+                            waiter.waitUntilStackCreateComplete(request -> request.stackName(stackName))
+                                .whenComplete((dsr, th) -> {
+                                    if (th != null) {
+                                        System.out.println("Error waiting for stack creation: " + th.getMessage());
+                                    } else {
+                                        dsr.matched().response().orElseThrow(() -> new RuntimeException("Failed to deploy"));
+                                        System.out.println("Stack created successfully");
+                                    }
+                                }).join();
+                        }
+                    } else {
+                        System.out.format("Error creating stack: " + t.getMessage(), t);
+                        throw new RuntimeException(t.getCause().getMessage(), t);
+                    }
+                }).join();
+        } else {
+            logger.info("{} stack already exists", CFN_TEMPLATE);
+        }
+    }
+
+    // Check to see if the Stack exists before deploying it
+
+    public static Boolean describeStack(String stackName) {
+        try {
+            CompletableFuture<?> future = getCloudFormationClient().describeStacks();
+            DescribeStacksResponse stacksResponse = (DescribeStacksResponse) future.join();
+            List<Stack> stacks = stacksResponse.stacks();
+            for (Stack myStack :stacks) {
+                if (myStack.stackName().compareTo(stackName)==0){
+                    return true;
+                }
+            }
+        } catch (CloudFormationException e) {
+            System.err.println(e.getMessage());
+        }
+        return false;
+    }
+    static void destroyCloudFormationStack(String stackName) {
+        getCloudFormationClient().deleteStack(b -> b.stackName(stackName))
+            .whenComplete((dsr, t) -> {
+                if (dsr != null) {
+                    System.out.println("Delete stack requested ....");
+                    try (CloudFormationAsyncWaiter waiter = getCloudFormationClient().waiter()) {
+                        waiter.waitUntilStackDeleteComplete(request -> request.stackName(stackName))
+                            .whenComplete((waiterResponse, throwable) ->
+                                System.out.println("Stack deleted successfully."))
+                            .join();
+                    }
+                } else {
+                    System.out.format("Error deleting stack: " + t.getMessage(), t);
+                    throw new RuntimeException(t.getCause().getMessage(), t);
+                }
+            }).join();
+    }
+
+    public static Map<String, String> getStackOutputs(String stackName) {
+        CloudFormationClient cfClient = CloudFormationClient.create();
+        DescribeStacksRequest describeStacksRequest = DescribeStacksRequest.builder()
```

Comment:
Why create a sync client. Use the static async client.

Reply:
Fixed - changed to Async